core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -343,16 +345,20 @@ class ReplicaFetcherThread(name: String,
       .filter { case (_, state) => state.isTruncatingLog }
       .map { case (tp, _) => tp -> epochCacheOpt(tp) }.toMap
 
-    val (partitionsWithEpoch, partitionsWithoutEpoch) = partitionEpochOpts.partition { case (_, epochCacheOpt) => epochCacheOpt.nonEmpty }
+    val (partitionsWithCache, partitionsWithoutCache) = partitionEpochOpts.partition { case (_, epochCacheOpt) => epochCacheOpt.nonEmpty }
+
+    debug(s"Build leaderEpoch request $partitionsWithCache")
```

Comment:
Since leader epoch request does not include partitions with undefined epoch anymore, should this log output `$partitionsWithoutEpoch` instead (and move debug statement below next statement?